### ISSUE TITLE
BRPOPLPUSH can totally block ioredis

### ISSUE THREAD

**Issue by jbeaudoin11** [luin/ioredis#1136]
Redis "B" commands like BRPOPLPUSH with an infinit wait, block ioredis' single connection which means no other calls to ioredis can go through.

Here is an example :
```javascript
// Create an interval refreshing a ttl key every sec
setInterval(async () => {
    await ioredis.expire("ttlKey", 60);
    //...
}, 1000);

//...

// Waiting to process the next message 
const value = await ioredis.brpoplpush("queue:a", "queue:b", 0);
//...
```
The `expire` command is always waiting for `brpoplpush` to resolve/reject.
But if there is no item in `queue:a`, we wait pretty much indefinitely. 
This is because ioredis has only one connection.

I'm not sure what i'm expecting from this github issue..
I don't think it's going to be "patch" in v4.
Maybe just to keep that in mind in v5.
Mean while I'm going to hack something out to make this work..

**Comment by alavers** [comment]
This is by design. You have to establish separate client connections for things like blocking commands and subscriptions. For example:

```
const Redis = require("ioredis");
const client = new Redis();
const blockingClient = new Redis(); 

blockingClient.brpoplpush('a1', 'a2', 0).then(console.log);
blockingClient.brpoplpush('b1', 'b2', 0).then(console.log);

client.rpush('a1', 'foo');
client.rpush('b1', 'bar');
```

**Comment by jbeaudoin11** [comment]
Yes i know it's "by design".

I guess I'm not clear in my original post.
I think you can see this as an argument to why ioredis should support connection pool out of the box. As a user of the lib, i don't know why it's not handle like in lots of other database libraries. Especially since redis has commands blocking the connection.

**Comment by denkquer** [comment]
Yeah in my opinion this should be alot clearer. I am just getting started with both redis and ioredis and getting no error message was very frustrating.

**Comment by denniske** [comment]
I just started with redis and was wondering why the first request took 10ms and subsequent took up to 5 seconds. Turns out I had a block 5000 call running in a loop and I was not aware that this would block this single connection.

So after creating like 3 more redis instances on different servers because I thought it is a network issue, I find out that this is by design. So well I am voting for a connection pool or at least a clear warning on e.g. the XREAD docs page.
